### PR TITLE
feat(nextjs): improve WithNxOptions typing

### PR DESCRIPTION
**Pull request by puku0x**
## Current Behavior

`WithNxOptions` does not extend `NextConfig`.

## Expected Behavior

```ts
import type { NextConfig } from 'next/dist/next-server/server/config';
import { WebpackConfigOptions } from '../src/utils/types';

export interface WithNxOptions extends NextConfig {
  nx?: WebpackConfigOptions;
}
```

```js
// eslint-disable-next-line @typescript-eslint/no-var-requires
const withNx = require('@nrwl/next/plugins/with-nx');

/**
 * @type {import('@nrwl/next/plugins/with-nx').WithNxOptions}
 **/
const nextConfig = {
  nx: {
    // Set this to false if you do not want to use SVGR
    // See: https://github.com/gregberge/svgr
    svgr: true,
  }
};

module.exports = withNx(nextConfig);
```
image

## Related Issue(s)

#5665

### CODE REVIEW

File: packages/next/plugins/with-nx.ts
Comment (puku0x) [inline packages/next/plugins/with-nx.ts:9]: This line can be removed because `NextConfig` already has it.

**Comment by puku0x** [comment]
Do I need to update the docs? I think I should wait for #6073 to be merged.

**Comment by puku0x** [comment]
@kirjai I added docs changes too. Could you check them too, please?

**Review by kirjai — CHANGES_REQUESTED** [review]
great improvement, thanks a lot! 

also really appreciate you staying on top of the docs changes as well.

---

i only have one small suggestion, regarding the docs specifically.
i see that you've made changes to both the pages in the `docs` directory, as well as `nx-dev/.../latest` directory. In this particular case, just the `docs` should be enough.

for more info, changes added to the `docs` will be deployed to the documentation on next release. whereas changes to `nx-dev` will show up in the docs "as soon as" this PR is merged. 

so, in this case, wee don't want to change docs in `nx-dev`, since this change to `next.config.js` is not going to be available to users until we do a new release.
does that make sense?

---

other than that, it all LGTM 👍

**Comment by puku0x** [comment]
Ah, I see. That's why the docs have two similar folders.

@kirjai  Thank you for your clarification. I will change this PR.

**Review by kirjai — APPROVED** [review]

**Pull request merged**
